process/peer/process.go

```diff
@@ -661,6 +657,24 @@ func (vs *validatorStatistics) ResetValidatorStatisticsAtNewEpoch(vInfos map[uin
 	return nil
 }
 
+func (vs *validatorStatistics) setToJailedIfNeeded(
+	peerAccount state.PeerAccountHandler,
+	validator *state.ValidatorInfo,
+) {
+	if !vs.flagJailedEnabled.IsSet() {
+		return
+	}
+
+	if validator.List == string(core.JailedList) && peerAccount.GetList() != string(core.JailedList) {
+		peerAccount.SetListAndIndex(validator.ShardId, string(core.JailedList), validator.Index)
+		return
+	}
+
+	if peerAccount.GetUnStakedEpoch() == core.DefaultUnstakedEpoch && peerAccount.GetList() == string(core.InactiveList) {
```

Comment:
no need for the validator.List == string (core.JailedList) ? Could it be not jailed in validator.List and still want to set it to jailed based on the new condition?

Reply:
This is for the scenario when in old code it was set to inactive, and should have been set to jailed